src/Appwrite/Platform/Modules/Sites/Http/Deployments/Download/Get.php

```diff
@@ -74,18 +78,33 @@ public function action(string $siteId, string $deploymentId, Response $response,
             throw new Exception(Exception::DEPLOYMENT_NOT_FOUND);
         }
 
-        $path = $deployment->getAttribute('path', '');
-        if (!$deviceForSites->exists($path)) {
-            throw new Exception(Exception::DEPLOYMENT_NOT_FOUND);
+        switch ($type) {
+            case 'output':
+                $build = $dbForProject->getDocument('builds', $deployment->getAttribute('buildId'));
+                if ($build->isEmpty()) {
+                    throw new Exception(Exception::BUILD_NOT_FOUND);
+                }
+
+                $path = $build->getAttribute('path', '');
+                $device = $deviceForBuilds;
+                break;
+            case 'source':
+                $path = $deployment->getAttribute('path', '');
+                $device = $deviceForSites;
+                break;
+        }
+
+        if (!$device->exists($path)) {
+            throw new Exception(Exception::BUILD_NOT_FOUND);
         }
 
         $response
             ->setContentType('application/gzip')
             ->addHeader('Expires', \date('D, d M Y H:i:s', \time() + (60 * 60 * 24 * 45)) . ' GMT') // 45 days cache
             ->addHeader('X-Peak', \memory_get_peak_usage())
-            ->addHeader('Content-Disposition', 'attachment; filename="' . $deploymentId . '.tar.gz"');
+            ->addHeader('Content-Disposition', 'attachment; filename="' . $deploymentId . '-' . $type . '.tar.gz"');
```

Comment:
Would there be any implications for updating this?

Reply:
![CleanShot 2025-03-04 at 14 29 52@2x](https://github.com/user-attachments/assets/b6567da2-7f95-48e5-804e-0159d30fca30)

As far as I can tell, the header is only used by browsers in naming newly downloaded file.